packages/headless/src/features/question-answering/question-answering-analytics-actions.ts

```diff
@@ -1,10 +1,31 @@
+import {CoveoSearchPageClient} from 'coveo.analytics';
 import {validatePayload} from '../../utils/validate-payload';
 import {AnalyticsType, makeAnalyticsAction} from '../analytics/analytics-utils';
 import {
   documentIdentifierPayloadDefinition,
   QuestionAnsweringDocumentIdActionCreatorPayload,
 } from './question-answering-document-id';
 
+type AnalyticsSmartSnippetFeedbackReason = Parameters<
+  CoveoSearchPageClient['logSmartSnippetFeedbackReason']
+>[0];
+
+export type SmartSnippetFeedback =
+  | 'does_not_answer'
+  | 'partially_answers'
+  | 'was_not_a_question';
+
+function parseSmartSnippetFeedbackReason(
+  feedback: SmartSnippetFeedback
+): AnalyticsSmartSnippetFeedbackReason {
+  switch (feedback) {
+    case 'does_not_answer':
+    case 'partially_answers':
+    case 'was_not_a_question':
+      return feedback as AnalyticsSmartSnippetFeedbackReason;
```

Comment:
shouldn't there be a default case here? and if not, what's the difference between doing this and simply 
```js
function parseSmartSnippetFeedbackReason(
  feedback: SmartSnippetFeedback
): AnalyticsSmartSnippetFeedbackReason {
      return feedback as AnalyticsSmartSnippetFeedbackReason;
}
```

Reply:
It's equivalent, yes. I forgot to leave a comment about this.

It's a leftover from when I tried to directly use the enum from `coveo.analytics.js`, but I got an error doing that since I was importing it from the `dist` folder. Upon changing course to instead return the string directly, I decided to keep this function for when the enum would be exported (which it is now, so I'll have to update this PR).